operator/controllers/btpoperator_controller.go

```diff
@@ -154,13 +390,20 @@ func (r *BtpOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	case "":
 		return ctrl.Result{}, r.HandleInitialState(ctx, cr)
 	case types.StateProcessing:
-		return ctrl.Result{RequeueAfter: ProcessingStateRequeueInterval}, r.HandleProcessingState(ctx, cr)
```

Comment:
we should determine whether update is required in `HandleProcessingState` and process the update as part of that method